worlds/rabi_ribi/__init__.py

```diff
@@ -76,27 +98,63 @@ def create_regions(self) -> None:
         Define regions and locations.
         This also defines access rules for the regions and locations.
         """
+        self.topology_present = bool(self.options.shuffle_map_transitions.value)
+
         region_def = RegionDef(self.multiworld, self.player, self.options)
+
+        # Universal tracker support, can be ignored for standard gen
+        if hasattr(self.multiworld, "re_gen_passthrough") and "Rabi-Ribi" in self.multiworld.re_gen_passthrough: # type: ignore
+            passthrough = self.multiworld.re_gen_passthrough["Rabi-Ribi"] # type: ignore
+            picked_templates = passthrough["picked_templates"]
+            map_transition_shuffle_order = passthrough["map_transition_shuffle_order"]
+            region_def.generate_set_seed(picked_templates, map_transition_shuffle_order)
+        else:
+            # Use standard generation
+            region_def.generate_seed()
+
         region_def.set_regions()
         region_def.connect_regions()
-        self.total_locations = region_def.set_locations(self.location_name_to_id)
+        self.total_locations = region_def.set_locations()
         region_def.set_events()
 
+        self.picked_templates = region_def.allocation.picked_templates
+        self.map_transition_shuffle_order = region_def.map_transition_shuffle_order
+
+        self.map_transition_shuffle_spoiler: List[str] = []
+        for (idx, x) in enumerate(self.map_transition_shuffle_order):
+            left = region_def.randomizer_data.walking_left_transitions[x]
+            right = region_def.randomizer_data.walking_right_transitions[idx]
+            left_name = convert_existing_rando_name_to_ap_name(left.origin_location)
+            right_name = convert_existing_rando_name_to_ap_name(right.origin_location)
+            self.map_transition_shuffle_spoiler.append(f'{left_name} -> {right_name}')
```

Comment:
Similar comment to the above. Would suggest putting this logic in a descriptive region_def method, and keeping this `create_region` method as just a simple wrapper for more complicated region logic.

Reply:
Makes sense